src/selected_words_counter/extract_files.py

```diff
@@ -86,19 +88,26 @@ def extract_msg_attachments(atarget_dir):
     # Extract all .msg files into a directory
     for afilepath in glob(atarget_dir + "*.msg"):
         afilepath = afilepath.replace("\\", "/")
-        try:
-            msg = extract_msg.Message(afilepath)
 
-            if len(msg.attachments) > 0:
-                a_output_directory = make_dir_from_filename(afilepath)
+        encodings = ["utf-8", "ISO-8859-1", "windows-1252"]
+        for encoding in encodings:
+            try:
+                msg = extract_msg.Message(afilepath, overrideEncoding=encoding)
 
-                for item in range(0, len(msg.attachments)):
-                    att = msg.attachments[item]
-                    msg.attachments[item].save(
-                        customPath=a_output_directory, customFilename=att.longFilename
-                    )
-        except Exception as e:
-            print(e)
+                if len(msg.attachments) > 0:
+                    a_output_directory = make_dir_from_filename(afilepath)
+
+                    for item in range(0, len(msg.attachments)):
+                        att = msg.attachments[item]
+                        msg.attachments[item].save(
+                            customPath=a_output_directory,
+                            customFilename=att.longFilename,
+                        )
+                return 0
+            except Exception as e:
+                continue
+        print(f"Failed to read {afilepath} with all tested encodings.")
```

Comment:
Maybe check in the main code where this function is called if it succeeeded? By the way returning 0 is usually not a sign that it was successfull. Maybe work with True and False instead? True meaning successful, False not successful.

Reply:
returning false now, it is a bit ugly like this indeed but it works will fix in a furter PR